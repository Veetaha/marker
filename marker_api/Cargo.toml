[package]
name = "marker_api"

categories  = ["development-tools"]
description = "Marker's API, designed for stability and usability"

edition    = { workspace = true }
keywords   = { workspace = true }
license    = { workspace = true }
repository = { workspace = true }
version    = { workspace = true }

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
visibility = { workspace = true }

[dev-dependencies]
expect-test = { workspace = true }

[features]
# Some items should only be used by the driver implementing the functionality,
# this feature enables the export of these items. Note that this interface is
# unstable. All usage of the driver api can change between releases.
driver-api = []
