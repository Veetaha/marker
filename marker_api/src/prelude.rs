//! This prelude is a collection of traits and types which are commonly used
//! when working with Marker. Simply add `use marker_api::prelude::*;` to your
//! file, to import them all.

// Traits:
pub use crate::ast::expr::ExprData;
pub use crate::ast::item::ItemData;
pub use crate::ast::pat::PatData;
pub use crate::ast::stmt::StmtData;
pub use crate::ast::ty::SynTyData;
pub use crate::diagnostic::EmissionNode;

// Common types
pub use crate::ast::expr::ExprKind;
pub use crate::ast::item::Body;
pub use crate::ast::item::ItemKind;
pub use crate::ast::pat::PatKind;
pub use crate::ast::stmt::StmtKind;
pub use crate::ast::ty::SemTyKind;
pub use crate::ast::ty::SynTyKind;
pub use crate::ast::Ident;
pub use crate::ast::Span;
pub use crate::AstContext;
