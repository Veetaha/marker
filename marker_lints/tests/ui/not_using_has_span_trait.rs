#![cfg_attr(
    marker,
    feature(register_tool),
    register_tool(marker),
    warn(marker::not_using_has_span_trait)
)]

use marker_api::prelude::*;

pub fn blackjack(_: u32, _: &Span<'_>) {}
pub fn rampage(_: &marker_api::span::Span<'_>, _: bool) {}

fn main() {}
