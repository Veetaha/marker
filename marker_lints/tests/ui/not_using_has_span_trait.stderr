warning: use impl HasSpan instead of Span for more flexibility
  --> $DIR/not_using_has_span_trait.rs:10:29
   |
10 | pub fn blackjack(_: u32, _: &Span<'_>) {}
   |                             ^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/not_using_has_span_trait.rs:5:10
   |
5  |     warn(marker::marker_lints::not_using_has_span_trait)
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use impl HasSpan instead of Span for more flexibility
  --> $DIR/not_using_has_span_trait.rs:11:19
   |
11 | pub fn rampage(_: &marker_api::span::Span<'_>, _: bool) {}
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 2 warnings emitted

